processing/data_collection/gazette/spiders/df_brasilia.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime
+import re
+
+import dateparser
+from scrapy import Request
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class DfBrasiliaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "5300108"
+    name = "df_brasilia"
+
+    GAZETTE_URL = "http://dodf.df.gov.br/listar"
+    DATE_REGEX = r"[0-9]{2}-[0-9]{2}-[0-9]{4}"
+    EXTRA_EDITION_TEXT = "EDICAO EXTR"
+    PDF_URL = "http://dodf.df.gov.br/index/visualizar-arquivo/?pasta={}&arquivo={}"
+
+    def start_requests(self):
+        """Requests page that has a list of all available years."""
+        yield Request(self.GAZETTE_URL, callback=self.parse_year_list)
+
+    def parse_year_list(self, response):
+        """Parses available years to request list of available months for each year."""
+        years_available = response.css(
+            "#local-arquivos .arquivo::attr(data-file)"
+        ).getall()
+
+        for year in years_available:
+            yield Request(
+                f"{self.GAZETTE_URL}?dir={year}",
+                meta={"year": year},
+                callback=self.parse_year,
+            )
+
+    def parse_year(self, response):
+        """Parses available months to request list of available dates for each month.
+        """
+        months_available = response.json().get("data", [])
+        year = response.meta["year"]
+
+        for month in months_available:
+            yield Request(
+                f"{self.GAZETTE_URL}?dir={year}/{month}",
+                meta=meta = {"month": month, "year": year},
+                callback=self.parse_month,
+            )
+
+    def parse_month(self, response):
+        """Parses available dates to request a list of documents for each date."""
+        month, year = response.meta["month"], response.meta["year"]
+        dates = response.json().get("data", [])
+
+        for gazette_name in dates.values():
+            url = f"{self.GAZETTE_URL}?dir={year}/{month}/{gazette_name}"
+            yield Request(url, callback=self.parse_gazette)
+
+    def parse_gazette(self, response):
+        """Parses list of documents to request each one for the date."""
+        json_response = response.json()
+
+        if not json_response:
+            self.logger.warning(f"Document not found in {response.url}")
+            return
```

Comment:
I'm thinking... if after you get the year and month data, you can still get no gazettes, why not try directly the month. AFAICS, the month request and year request are almost the same. They just has a extra meta value. So, why not request the complete date upfront?

I do not know if I`m missing something. I'm just sharing a though that I have while I was reading the source code.  I could not check the web site in depth to validate my idea.

Reply:
Requesting the months upfront would be possible, but we'd have to revert to using hardcoded stuff or some lib with a dictionary since the format of the url is the following: `http://dodf.df.gov.br/listar?dir=2001/11_Novembro`.

I thought it wasn't worth the trouble since those are just the initial requests and are so few. What do you think?

For the complete date I couldn't find any pattern since it uses the "name" of the edition and I think that's manually done.

Reply:
Oh, no no. I'm not referencing to the editions "directories". I'm just referencing to request where you ask for the moth "directory" **containing** the editions. In other words, to generate the request for URL like this `http://dodf.df.gov.br/listar?dir=2001/11_Novembro` you probably do not need to request the years "directories". 

This is not so important. I would like just to check if my thoughts make sense or not.